core/test/allocation_test.jl

```diff
@@ -75,7 +78,14 @@ end
     @test objective isa JuMP.AffExpr # Affine expression
     @test :F_abs in keys(problem.obj_dict)
     F_abs = problem[:F_abs]
-    @test objective == F_abs[NodeID(5)] + F_abs[NodeID(6)]
+    @test string(objective) == string(
```

Comment:
Both these string representation tests are still failing. What is the reason you moved away from testing the objective function object directly?

Reply:
Before the objective function only had trivial coefficients (namely 1). With non-trivial float coefficients I could not get equality tests to pass on the object itself, I'll have another look at it.

Reply:
I think the tests fail because the order of the terms in the expression is not constant. I'll test it in a different way